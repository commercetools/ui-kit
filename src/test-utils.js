import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider, addLocaleData } from 'react-intl';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import en from 'react-intl/locale-data/en';
import de from 'react-intl/locale-data/de';
import es from 'react-intl/locale-data/es';
import messages from '../i18n/core.json';

addLocaleData(en);
addLocaleData(de);
addLocaleData(es);

const customRender = (
  node,
  {
    locale = 'en', // react-router
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
    ...rtlOptions
  } = {}
) => ({
  ...render(
    <IntlProvider locale={locale} messages={messages}>
      <Router history={history}>{node}</Router>
    </IntlProvider>,
    rtlOptions
  ),
  // adding `history` to the returned utilities to allow us
  // to reference it in our tests (just try to avoid using
  // this to test implementation details).
  history,
});

// re-export everything
export * from 'react-testing-library';

// override render method
export { customRender as render };
