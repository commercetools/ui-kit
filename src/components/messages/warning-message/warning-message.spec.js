import React from 'react';
import { render } from '../../../test-utils';
import ErrorMessage from './warning-message';

it('should render children', () => {
  const { container } = render(
    <ErrorMessage>Some warning message</ErrorMessage>
  );

  expect(container).toHaveTextContent('Some warning message');
});
