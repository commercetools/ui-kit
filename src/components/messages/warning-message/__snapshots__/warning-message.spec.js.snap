// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should match snapshot 1`] = `
<TextDetail
  tone="warning"
>
  <div>
    hi
  </div>
</TextDetail>
`;
